File: check-translations.py
import re

with open('./src/i18n/ui.ts', 'r', encoding='utf-8') as f:
    lines = f.readlines()

# Extract English keys
en_keys = []
in_en_section = False
for line in lines:
    line = line.strip()
    if line == 'en: {':
        in_en_section = True
    elif line == '},':
        in_en_section = False
    elif in_en_section and ':' in line and not line.endswith(','):
        # Extract key name before colon
        key = line.split(':')[0].strip()
        # Only add if key is a valid identifier
        if re.match(r'^[a-zA-Z0-9_]+$', key):
            en_keys.append(key)

# Extract Chinese keys
zh_keys = []
in_zh_section = False
for line in lines:
    line = line.strip()
    if line == "'zh-CN': {":
        in_zh_section = True
    elif line == '},':
        in_zh_section = False
    elif in_zh_section and ':' in line and not line.endswith(','):
        # Extract key name before colon
        key = line.split(':')[0].strip()
        # Only add if key is a valid identifier
        if re.match(r'^[a-zA-Z0-9_]+$', key):
            zh_keys.append(key)

print(f'English keys: {len(en_keys)}')
print(f'Chinese keys: {len(zh_keys)}')

# Find missing keys in Chinese
missing_in_zh = [key for key in en_keys if key not in zh_keys]
if missing_in_zh:
    print('\nMissing in Chinese:')
    for key in missing_in_zh:
        print(f'- {key}')

# Find extra keys in Chinese
extra_in_zh = [key for key in zh_keys if key not in en_keys]
if extra_in_zh:
    print('\nExtra in Chinese:')
    for key in extra_in_zh:
        print(f'- {key}')

if not missing_in_zh and not extra_in_zh:
    print('\nAll keys are matched!')

File: check-translations.js
import fs from 'fs';

const content = fs.readFileSync('./src/i18n/ui.ts', 'utf8');

// Extract the translations object
const translationsMatch = content.match(/const translations = (\{[\s\S]*?\}) as const/);
if (!translationsMatch) {
  console.error('Could not find translations object');
  process.exit(1);
}

// Parse the translations object
let translationsStr = translationsMatch[1];
// Add quotes around keys that don't have them
translationsStr = translationsStr.replace(/([a-zA-Z0-9_]+):/g, '"$1":');
// Replace single quotes with double quotes
translationsStr = translationsStr.replace(/'/g, '"');
// Parse as JSON
const translations = JSON.parse(translationsStr);

const enKeys = Object.keys(translations.en);
const zhKeys = Object.keys(translations['zh-CN']);

console.log('English keys:', enKeys.length);
console.log('Chinese keys:', zhKeys.length);

const missingInZh = enKeys.filter(key => !zhKeys.includes(key));
const extraInZh = zhKeys.filter(key => !enKeys.includes(key));

if (missingInZh.length > 0) {
  console.log('\nMissing in Chinese:');
  missingInZh.forEach(key => console.log(`- ${key}`));
}

if (extraInZh.length > 0) {
  console.log('\nExtra in Chinese:');
  extraInZh.forEach(key => console.log(`- ${key}`));
}

if (missingInZh.length === 0 && extraInZh.length === 0) {
  console.log('\nAll keys are matched!');
}
